str/data_frame_creator.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+# pylint: disable=import-error, too-many-locals, broad-exception-raised
+"""
+This script merges all the VCFs from one STR caller into a .CSV (ExpansionHunter) or .TSV (GangSTR) format that can be read into R. 
+analysis-runner --access-level test --dataset tob-wgs --description 'data frame creator' --output-dir 'hoptan-str/tob_test_crams/data_frames' data_frame_creator.py  --input-dir-eh=gs://cpg-tob-wgs-test/hoptan-str/tob_test_crams/output_calls/eh_0_based --input-dir-gangstr=gs://cpg-tob-wgs-test/hoptan-str/tob_test_crams/output_calls/gangstr_0_based --output-name-eh=eh.csv --output-name-gangstr=gangstr.tsv
+
+"""
+import click
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import remote_tmpdir, output_path
+import hailtop.batch as hb
+from google.cloud import storage
+
+config = get_config()
+
+
+def eh_csv_writer(input_dir):
+    """Creates a CSV file containing dataframe of merged EH VCFs"""
+    bucket_name, *components = input_dir[5:].split('/')
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+    blobs = client.list_blobs(bucket_name, prefix='/'.join(components))
+    files = {f'gs://{bucket_name}/{blob.name}' for blob in blobs}
```

Comment:
```suggestion
    files = to_path(input_dir).glob('*.vcf')
```

this is using `cloudpathlib.AnyPath/CloudPath` or `cpg_utils.to_path`. This also removes the check you have a few lines later where you only consider VCF files from the list of returned blobs

You can also open and read these results directly

```
for file in files:
    with file.open() as f:
         ...
```

There are reasons to avoid using cloudpathlib methods like this, but for your purpose here we are trying to retrieve and read all contents in the folder, so I don't think the downsides are super relevant

Reply:
This is neat. I think I was hesitating to use `cloudpathlib` because it's harder to see what's happening underneath the hood, and was OK with the chunkier method as long as it was working. Will use `to_path` from now on.